### PR TITLE
Added Check Style

### PR DESCRIPTION
**Pull request by oblodgett**
This PR add check style which checks the formatting of the code. This will alert to extraneous space in the code or spaces vs tabs, etc. There is 3 major changes (although minor in the grand scheme of things)
1. Single line blocks will need to be enclosed with curly braces and have proper spacing Shift - Command - F on mac will do this.
2. Arrays can't end with a trailing comma even though java allows it
3. Util classes (static classes) have to have a private (hidden) constructor (this is fairly rare in the codebase)

### CODE REVIEW

File: src/main/java/org/alliancegenome/curation_api/jobs/executors/AgmDiseaseAnnotationExecutor.java [hunk 1]
```diff
@@ -36,28 +33,33 @@ public void execLoad(BulkLoadFile bulkLoadFile, Boolean cleanUp) {
...
-		
+
 		bulkLoadFile.setRecordCount(annotations.size() + bulkLoadFile.getRecordCount());
 		bulkLoadFileDAO.merge(bulkLoadFile);
 
 		BulkLoadFileHistory history = new BulkLoadFileHistory(annotations.size());
 		createHistory(history, bulkLoadFile);
 		boolean success = runLoad(agmDiseaseAnnotationService, history, dataProvider, annotations, annotationIdsLoaded);
-		if(success && cleanUp) runCleanup(diseaseAnnotationService, history, dataProvider.name(), annotationIdsBefore, annotationIdsLoaded, "AGM disease annotation", bulkLoadFile.getMd5Sum());
+		if(success && cleanUp) {
```
Comment (mluypaert) [inline src/main/java/org/alliancegenome/curation_api/jobs/executors/AgmDiseaseAnnotationExecutor.java:58]: 🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAroundCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'if' is not followed by whitespace.

File: src/main/java/org/alliancegenome/curation_api/jobs/executors/associations/alleleAssociations/AlleleGeneAssociationExecutor.java [hunk 2]
```diff
@@ -21,36 +21,40 @@
...
-		
+
 		bulkLoadFile.setRecordCount(associations.size() + bulkLoadFile.getRecordCount());
 		bulkLoadFileDAO.merge(bulkLoadFile);
 
 		BulkLoadFileHistory history = new BulkLoadFileHistory(associations.size());
 		createHistory(history, bulkLoadFile);
 		boolean success = runLoad(alleleGeneAssociationService, history, dataProvider, associations, associationIdsLoaded);
-		if(success && cleanUp) runCleanup(alleleGeneAssociationService, history, dataProvider.name(), associationIdsBefore, associationIdsLoaded, bulkLoadFile.getMd5Sum());
+		if(success && cleanUp) {
```
Comment (mluypaert) [inline src/main/java/org/alliancegenome/curation_api/jobs/executors/associations/alleleAssociations/AlleleGeneAssociationExecutor.java:55]: 🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAroundCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'if' is not followed by whitespace.

File: src/main/java/org/alliancegenome/curation_api/jobs/executors/AlleleExecutor.java [hunk 3]
```diff
@@ -19,38 +19,42 @@
...
+
 		bulkLoadFile.setRecordCount(alleles.size() + bulkLoadFile.getRecordCount());
 		bulkLoadFileDAO.merge(bulkLoadFile);
-		
+
 		BulkLoadFileHistory history = new BulkLoadFileHistory(alleles.size());
 		createHistory(history, bulkLoadFile);
 		boolean success = runLoad(alleleService, history, dataProvider, alleles, alleleIdsLoaded);
-		if(success && cleanUp) runCleanup(alleleService, history, bulkLoadFile, alleleIdsBefore, alleleIdsLoaded);
+		if(success && cleanUp) {
```
Comment (mluypaert) [inline src/main/java/org/alliancegenome/curation_api/jobs/executors/AlleleExecutor.java:55]: 🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAroundCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'if' is not followed by whitespace.

File: src/main/java/org/alliancegenome/curation_api/jobs/executors/ConstructExecutor.java [hunk 4]
```diff
@@ -24,42 +24,45 @@
...
+
 		bulkLoadFile.setRecordCount(constructs.size() + bulkLoadFile.getRecordCount());
 		bulkLoadFileDAO.merge(bulkLoadFile);
-		
+
 		BulkLoadFileHistory history = new BulkLoadFileHistory(constructs.size());
 		createHistory(history, bulkLoadFile);
 		boolean success = runLoad(constructService, history, dataProvider, constructs, constructIdsLoaded);
-		if(success && cleanUp) runCleanup(constructService, history, dataProvider.name(), constructIdsBefore, constructIdsLoaded, bulkLoadFile.getMd5Sum());
+		if(success && cleanUp) {
```
Comment (mluypaert) [inline src/main/java/org/alliancegenome/curation_api/jobs/executors/ConstructExecutor.java:63]: 🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAroundCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'if' is not followed by whitespace.

File: src/main/java/org/alliancegenome/curation_api/jobs/executors/GeneDiseaseAnnotationExecutor.java [hunk 5]
```diff
@@ -36,25 +33,31 @@ public void execLoad(BulkLoadFile bulkLoadFile, Boolean cleanUp) {
...
+
+		bulkLoadFile.setRecordCount(annotations.size() + bulkLoadFile.getRecordCount());
 		bulkLoadFileDAO.merge(bulkLoadFile);
-		
+
 		BulkLoadFileHistory history = new BulkLoadFileHistory(annotations.size());
 		createHistory(history, bulkLoadFile);
 		boolean success = runLoad(geneDiseaseAnnotationService, history, dataProvider, annotations, annotationIdsLoaded);
-		if(success && cleanUp) runCleanup(diseaseAnnotationService, history, dataProvider.name(), annotationIdsBefore, annotationIdsLoaded, "gene disease annotation", bulkLoadFile.getMd5Sum());
+		if(success && cleanUp) {
```
Comment (mluypaert) [inline src/main/java/org/alliancegenome/curation_api/jobs/executors/GeneDiseaseAnnotationExecutor.java:58]: 🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAroundCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'if' is not followed by whitespace.

File: src/main/java/org/alliancegenome/curation_api/jobs/executors/AgmExecutor.java [hunk 6]
```diff
@@ -48,18 +49,19 @@ public void execLoad(BulkLoadFile bulkLoadFile, Boolean cleanUp) {
...
+
 		bulkLoadFile.setRecordCount(agms.size() + bulkLoadFile.getRecordCount());
 		bulkLoadFileDAO.merge(bulkLoadFile);
-		
+
 		BulkLoadFileHistory history = new BulkLoadFileHistory(agms.size());
 		createHistory(history, bulkLoadFile);
 		boolean success = runLoad(affectedGenomicModelService, history, dataProvider, agms, agmIdsLoaded);
-		if(success && cleanUp) runCleanup(affectedGenomicModelService, history, bulkLoadFile, agmIdsBefore, agmIdsLoaded);
+		if(success && cleanUp) {
```
Comment (mluypaert) [inline src/main/java/org/alliancegenome/curation_api/jobs/executors/AgmExecutor.java:59]: 🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAroundCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'if' is not followed by whitespace.

File: src/main/java/org/alliancegenome/curation_api/jobs/executors/GeneExecutor.java [hunk 7]
```diff
@@ -37,25 +34,31 @@ public void execLoad(BulkLoadFile bulkLoadFile, Boolean cleanUp) {
...
-		
+
 		bulkLoadFile.setRecordCount(genes.size() + bulkLoadFile.getRecordCount());
 		bulkLoadFileDAO.merge(bulkLoadFile);
 
 		BulkLoadFileHistory history = new BulkLoadFileHistory(genes.size());
 		createHistory(history, bulkLoadFile);
 		boolean success = runLoad(geneService, history, dataProvider, genes, geneIdsLoaded);
-		if(success && cleanUp) runCleanup(geneService, history, bulkLoadFile, geneIdsBefore, geneIdsLoaded);
+		if(success && cleanUp) {
```
Comment (mluypaert) [inline src/main/java/org/alliancegenome/curation_api/jobs/executors/GeneExecutor.java:59]: 🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAroundCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'if' is not followed by whitespace.

File: src/main/java/org/alliancegenome/curation_api/jobs/executors/VariantExecutor.java [hunk 8]
```diff
@@ -19,42 +19,45 @@
...
+
 		bulkLoadFile.setRecordCount(variants.size() + bulkLoadFile.getRecordCount());
 		bulkLoadFileDAO.merge(bulkLoadFile);
-		
+
 		BulkLoadFileHistory history = new BulkLoadFileHistory(variants.size());
 		createHistory(history, bulkLoadFile);
 		boolean success = runLoad(variantService, history, dataProvider, variants, variantIdsLoaded);
-		if(success && cleanUp) runCleanup(variantService, history, bulkLoadFile, variantIdsBefore, variantIdsLoaded);
+		if(success && cleanUp) {
```
Comment (mluypaert) [inline src/main/java/org/alliancegenome/curation_api/jobs/executors/VariantExecutor.java:55]: 🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAroundCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'if' is not followed by whitespace.

File: src/main/java/org/alliancegenome/curation_api/jobs/executors/AlleleDiseaseAnnotationExecutor.java [hunk 9]
```diff
@@ -36,25 +33,31 @@ public void execLoad(BulkLoadFile bulkLoadFile, Boolean cleanUp) {
...
-		
+
 		bulkLoadFile.setRecordCount(annotations.size() + bulkLoadFile.getRecordCount());
 		bulkLoadFileDAO.merge(bulkLoadFile);
 
 		BulkLoadFileHistory history = new BulkLoadFileHistory(annotations.size());
 		createHistory(history, bulkLoadFile);
 		boolean success = runLoad(alleleDiseaseAnnotationService, history, dataProvider, annotations, annotationIdsLoaded);
-		if(success && cleanUp) runCleanup(diseaseAnnotationService, history, dataProvider.name(), annotationIdsBefore, annotationIdsLoaded, "allele disease annotation", bulkLoadFile.getMd5Sum());
+		if(success && cleanUp) {
```
Comment (mluypaert) [inline src/main/java/org/alliancegenome/curation_api/jobs/executors/AlleleDiseaseAnnotationExecutor.java:58]: 🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAroundCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'if' is not followed by whitespace.

File: src/main/java/org/alliancegenome/curation_api/jobs/executors/associations/alleleAssociations/AlleleGeneAssociationExecutor.java (hunk 2 above)
Comment (devopsagr) [inline src/main/java/org/alliancegenome/curation_api/jobs/executors/associations/alleleAssociations/AlleleGeneAssociationExecutor.java]: 🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAroundCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'if' is not followed by whitespace.

**Comment by oblodgett** [comment]
@adamgibs this is only for *.java files. I would like to set this up for *.js files but the rules would be different. Maybe we can find an example checkstyle.xml for *.js?

**Review by abecerra — APPROVED** [review]
Could you add to the README a line explaining how to use it locally before pushing?

**Comment by adamgibs** [comment]
@oblodgett It looks like checkstyle is specifically for Java. I'm not seeing any examples of using with with JavaScript files.

File: src/main/java/org/alliancegenome/curation_api/auth/AuthenticationFilter.java [hunk 10]
```diff
@@ -41,38 +41,27 @@
...
-	Event<Person> userAuthenticatedEvent;
+	@AuthenticatedUser Event<Person> userAuthenticatedEvent;
 
-	@Inject
-	AuthenticationService authenticationService;
+	@Inject AuthenticationService authenticationService;
 
-	@Inject
-	PersonDAO personDAO;
+	@Inject PersonDAO personDAO;
```
Comment (cmpich) [inline src/main/java/org/alliancegenome/curation_api/auth/AuthenticationFilter.java:48]: I like to have the annotation on top of the line it's annotating as it is easier to read for me. 
Can we not change this?

File: src/main/java/org/alliancegenome/curation_api/model/entities/Allele.java [hunk 11]
```diff
@@ -124,7 +155,13 @@ public class Allele extends GenomicEntity {
 	@JsonView({ View.FieldsOnly.class })
 	private AlleleGermlineTransmissionStatusSlotAnnotation alleleGermlineTransmissionStatus;
 
-	@IndexedEmbedded(includePaths = { "functionalImpacts.name", "phenotypeTerm.curie", "phenotypeTerm.name", "phenotypeStatement","evidence.curie", "functionalImpacts.name_keyword", "phenotypeTerm.curie_keyword", "phenotypeTerm.name_keyword", "phenotypeStatement_keyword", "evidence.curie_keyword"})
+	@IndexedEmbedded(
+		includePaths = {
+			"functionalImpacts.name", "phenotypeTerm.curie", "phenotypeTerm.name", "phenotypeStatement", "evidence.curie",
+			"functionalImpacts.name_keyword", "phenotypeTerm.curie_keyword", "phenotypeTerm.name_keyword", "phenotypeStatement_keyword",
```
Comment (cmpich) [inline src/main/java/org/alliancegenome/curation_api/model/entities/Allele.java:161]: Nice!

**Pull request merged**